diff -Naur a/Makefile.msc b/Makefile.msc
--- a/Makefile.msc	2026-07-02 20:53:55.000000000 +0200
+++ b/Makefile.msc	2026-08-04 12:33:34.244926174 +0200
@@ -1242,9 +1242,9 @@
 # If either debugging or symbols are enabled, enable PDBs.
 #
 !IF $(DEBUG)>1 || $(SYMBOLS)!=0
-LDFLAGS = /NODEFAULTLIB:msvcrt /DEBUG $(LDOPTS)
+LDFLAGS = /DEBUG $(LDOPTS)
 !ELSE
-LDFLAGS = /NODEFAULTLIB:msvcrt $(LDOPTS)
+LDFLAGS = $(LDOPTS)
 !ENDIF
 
 # <<mark>>
