diff -Naur a/qtbase/cmake/FindWrapSystemDoubleConversion.cmake b/qtbase/cmake/FindWrapSystemDoubleConversion.cmake
--- a/qtbase/cmake/FindWrapSystemDoubleConversion.cmake	2025-06-09 15:43:42.000000000 +0200
+++ b/qtbase/cmake/FindWrapSystemDoubleConversion.cmake	2025-10-16 14:40:49.027418131 +0200
@@ -14,7 +14,7 @@
 # Upstream can be built either with CMake and then provides a Config file, or with Scons in which
 # case there's no Config file.
 # A Find module might be provided by a 3rd party, for example Conan might generate a Find module.
-find_package(double-conversion ${${CMAKE_FIND_PACKAGE_NAME}_FIND_VERSION} QUIET)
+find_package(double-conversion QUIET)
 set(__double_conversion_target_name "double-conversion::double-conversion")
 if(double-conversion_FOUND AND TARGET "${__double_conversion_target_name}")
     set(__double_conversion_found TRUE)
diff -Naur a/qtbase/cmake/FindWrapSystemFreetype.cmake b/qtbase/cmake/FindWrapSystemFreetype.cmake
--- a/qtbase/cmake/FindWrapSystemFreetype.cmake	2025-06-09 15:43:42.000000000 +0200
+++ b/qtbase/cmake/FindWrapSystemFreetype.cmake	2025-10-16 14:41:09.376023784 +0200
@@ -11,10 +11,10 @@
 
 # Hunter has the package named freetype, but exports the Freetype::Freetype target as upstream
 # First try the CONFIG package, and afterwards the MODULE if not found
-find_package(Freetype ${${CMAKE_FIND_PACKAGE_NAME}_FIND_VERSION}
+find_package(Freetype
              CONFIG NAMES Freetype freetype QUIET)
 if(NOT Freetype_FOUND)
-    find_package(Freetype ${${CMAKE_FIND_PACKAGE_NAME}_FIND_VERSION} QUIET MODULE)
+    find_package(Freetype QUIET MODULE)
 endif()
 
 if(Freetype_FOUND)
diff -Naur a/qtbase/cmake/FindWrapSystemHarfbuzz.cmake b/qtbase/cmake/FindWrapSystemHarfbuzz.cmake
--- a/qtbase/cmake/FindWrapSystemHarfbuzz.cmake	2025-06-09 15:43:42.000000000 +0200
+++ b/qtbase/cmake/FindWrapSystemHarfbuzz.cmake	2025-10-16 14:41:19.274697179 +0200
@@ -9,7 +9,7 @@
 endif()
 set(WrapSystemHarfbuzz_REQUIRED_VARS __harfbuzz_found)
 
-find_package(harfbuzz ${${CMAKE_FIND_PACKAGE_NAME}_FIND_VERSION} QUIET)
+find_package(harfbuzz QUIET)
 
 # Gentoo has some buggy version of a harfbuzz Config file. Check if include paths are valid.
 set(__harfbuzz_target_name "harfbuzz::harfbuzz")
diff -Naur a/qtbase/cmake/FindWrapSystemJpeg.cmake b/qtbase/cmake/FindWrapSystemJpeg.cmake
--- a/qtbase/cmake/FindWrapSystemJpeg.cmake	2025-06-09 15:43:42.000000000 +0200
+++ b/qtbase/cmake/FindWrapSystemJpeg.cmake	2025-10-16 14:41:29.691387551 +0200
@@ -7,7 +7,7 @@
 endif()
 set(WrapSystemJpeg_REQUIRED_VARS __jpeg_found)
 
-find_package(JPEG ${${CMAKE_FIND_PACKAGE_NAME}_FIND_VERSION} QUIET)
+find_package(JPEG QUIET)
 
 set(__jpeg_target_name "JPEG::JPEG")
 if(JPEG_FOUND AND TARGET "${__jpeg_target_name}")
diff -Naur a/qtbase/cmake/FindWrapSystemMd4c.cmake b/qtbase/cmake/FindWrapSystemMd4c.cmake
--- a/qtbase/cmake/FindWrapSystemMd4c.cmake	2025-06-09 15:43:42.000000000 +0200
+++ b/qtbase/cmake/FindWrapSystemMd4c.cmake	2025-10-16 14:42:24.030762492 +0200
@@ -7,7 +7,7 @@
 endif()
 set(WrapSystemMd4c_REQUIRED_VARS __md4c_found)
 
-find_package(md4c ${${CMAKE_FIND_PACKAGE_NAME}_FIND_VERSION} QUIET)
+find_package(md4c QUIET)
 
 set(__md4c_target_name "md4c::md4c")
 
diff -Naur a/qtbase/cmake/FindWrapSystemPCRE2.cmake b/qtbase/cmake/FindWrapSystemPCRE2.cmake
--- a/qtbase/cmake/FindWrapSystemPCRE2.cmake	2025-06-09 15:43:42.000000000 +0200
+++ b/qtbase/cmake/FindWrapSystemPCRE2.cmake	2025-10-16 14:41:39.252966319 +0200
@@ -7,7 +7,7 @@
 endif()
 set(WrapSystemPCRE2_REQUIRED_VARS __pcre2_found)
 
-find_package(PCRE2 ${${CMAKE_FIND_PACKAGE_NAME}_FIND_VERSION} COMPONENTS 16BIT QUIET)
+find_package(PCRE2 COMPONENTS 16BIT QUIET)
 
 set(__pcre2_target_name "PCRE2::16BIT")
 if(PCRE2_FOUND AND TARGET "${__pcre2_target_name}")
diff -Naur a/qtbase/cmake/FindWrapSystemPNG.cmake b/qtbase/cmake/FindWrapSystemPNG.cmake
--- a/qtbase/cmake/FindWrapSystemPNG.cmake	2025-06-09 15:43:42.000000000 +0200
+++ b/qtbase/cmake/FindWrapSystemPNG.cmake	2025-10-16 14:41:52.178601288 +0200
@@ -9,7 +9,7 @@
 endif()
 set(WrapSystemPNG_REQUIRED_VARS __png_found)
 
-find_package(PNG ${${CMAKE_FIND_PACKAGE_NAME}_FIND_VERSION} QUIET)
+find_package(PNG QUIET)
 
 set(__png_target_name "PNG::PNG")
 if(PNG_FOUND AND TARGET "${__png_target_name}")
diff -Naur a/qtbase/cmake/FindWrapSystemZLIB.cmake b/qtbase/cmake/FindWrapSystemZLIB.cmake
--- a/qtbase/cmake/FindWrapSystemZLIB.cmake	2025-06-09 15:43:42.000000000 +0200
+++ b/qtbase/cmake/FindWrapSystemZLIB.cmake	2025-10-16 14:42:00.796486376 +0200
@@ -10,7 +10,7 @@
 
 set(WrapSystemZLIB_FOUND OFF)
 
-find_package(ZLIB ${${CMAKE_FIND_PACKAGE_NAME}_FIND_VERSION})
+find_package(ZLIB)
 
 if(ZLIB_FOUND)
     set(WrapSystemZLIB_FOUND ON)
diff -Naur a/qtquick3d/cmake/FindWrapSystemOpenXR.cmake b/qtquick3d/cmake/FindWrapSystemOpenXR.cmake
--- a/qtquick3d/cmake/FindWrapSystemOpenXR.cmake	2025-07-06 08:09:17.000000000 +0200
+++ b/qtquick3d/cmake/FindWrapSystemOpenXR.cmake	2025-10-16 14:42:10.106904785 +0200
@@ -65,7 +65,7 @@
 endif ()
 
 
-find_package(OpenXR ${${CMAKE_FIND_PACKAGE_NAME}_FIND_VERSION} QUIET)
+find_package(OpenXR QUIET)
 
 set(__openxr_target_name "OpenXR::openxr_loader")
 if(OpenXR_FOUND AND TARGET "${__openxr_target_name}")
