diff --git a/libnotify/notification.c b/libnotify/notification.c
index 1504c8f..a6f30ef 100644
--- a/libnotify/notification.c
+++ b/libnotify/notification.c
@@ -1034,6 +1034,10 @@ add_portal_notification (GDBusProxy         *proxy,
                         GVariantBuilder button;
                         ActionInfo *ai = g_ptr_array_index (priv->actions, i);
 
+                        if (g_ascii_strcasecmp (ai->id, "default") == 0) {
+                            continue;
+                        }
+
                         g_variant_builder_init (&button, G_VARIANT_TYPE_VARDICT);
 
                         g_variant_builder_add (&button, "{sv}", "action",
