diff --git a/CMakeLists.txt b/CMakeLists.txt
index 747e4c05..a33d5c23 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,8 +15,11 @@ set(QCA_LIB_MAJOR_VERSION "2")
 set(QCA_LIB_MINOR_VERSION "3")
 set(QCA_LIB_PATCH_VERSION "9")
 
+if (NOT DEFINED CMAKE_MACOSX_RPATH)
+    set(CMAKE_MACOSX_RPATH FALSE)
+endif()
 if(POLICY CMP0042)
-  cmake_policy(SET CMP0042 OLD)
+    cmake_policy(SET CMP0042 NEW)
 endif()
 
 option(BUILD_WITH_QT6 "Build with Qt 6" OFF)
