diff --git a/CMakeLists.txt b/CMakeLists.txt
index c4a7eff..05bf732 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,7 +27,6 @@ set(QtWebDAV_SOURCES
 set(QtWebDAV_HEADERS
     qnaturalsort.h
     qwebdav.h
-    qwebdav_global.h
     qwebdavdirparser.h
     qwebdavitem.h
 )
@@ -63,7 +62,7 @@ target_include_directories(QtWebDAV PUBLIC
 target_compile_definitions(QtWebDAV PRIVATE DEBUG_WEBDAV)
 
 # Link Qt libraries
-target_link_libraries(QtWebDAV PUBLIC
+target_link_libraries(QtWebDAV PRIVATE
     Qt${QT_VERSION_MAJOR}::Network
     Qt${QT_VERSION_MAJOR}::Xml
 )
