diff --git a/pjsip/src/pjsua-lib/pjsua_pres.c b/pjsip/src/pjsua-lib/pjsua_pres.c
index 7df0075..483fcf2 100644
--- a/pjsip/src/pjsua-lib/pjsua_pres.c
+++ b/pjsip/src/pjsua-lib/pjsua_pres.c
@@ -2217,7 +2217,7 @@ static void subscribe_buddy(pjsua_buddy_id buddy_id,
         return;
     }

-    pjsip_dlg_dec_lock(buddy->dlg);
+    pjsip_dlg_dec_lock(dlg);
     if (tmp_pool) pj_pool_release(tmp_pool);
     pj_log_pop_indent();
 }
